src/azure-cli/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -955,6 +972,13 @@ def _get_event_subscription_info(    # pylint: disable=too-many-locals,too-many-
         tennant_id = azure_active_directory_tenant_id
         application_id = azure_active_directory_application_id_or_uri
 
+    if endpoint_type.lower() != STORAGEQUEUE_DESTINATION.lower() and \
+       storage_queue_msg_ttl is not None:
+        raise CLIError('usage error: --storage-queue-msg-ttl is only applicable for endpoint type StorageQueue.')
+
+    if storage_queue_msg_ttl is not None:
+        storage_queue_msg_ttl = int(storage_queue_msg_ttl)
```

Comment:
See https://github.com/Azure/azure-cli/pull/17440#discussion_r600999064

Reply:
is type=int nullable though?

Reply:
Yes. If customers didn't use this parameter, then it's `None`.